### PR TITLE
APPCAFE-06: Fetch data from firebase to the app

### PR DESCRIPTION
**Pull request by 1699kratos**
## ISSUE:
* https://trello.com/c/85elcKa4

## FIX:
* Pods from firebase installed
* Pods for SDWebImage installed
* Changed a layout from menu item xib file
* Created own function to don´t need other packages, made it in uikit and swiftui to download the image

## EVIDENCES

https://user-images.githubusercontent.com/49655011/143660123-7c1006c6-83f5-4c6c-9817-edbb67c985c3.mp4

image

image

### CODE REVIEW

File: CafePrices/CafePrices/Controller/CafeteriasViewController.swift [hunk 1]
```diff
@@ -44,17 +45,19 @@ extension CafeteriasViewController: UITableViewDataSource, UITableViewDelegate {
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "cafeteria", for: indexPath) as? TableViewCell ?? TableViewCell()
-        cell.cafeteriaName.text = cafes[indexPath.row].cafeteriaName
-        let fileUrl = URL(string: cafes[indexPath.row].cafeteriaImage)
-        cell.cafeteriaImage.sd_setImage(with: fileUrl)
+        cell.cafeteriaName.text = cafes[indexPath.row].name
+        let fileUrl = URL(string: cafes[indexPath.row].image)
+        cell.cafeteriaImage.load(url: fileUrl!)
```
Comment (hachiosaki) [inline CafePrices/CafePrices/Controller/CafeteriasViewController.swift]: don't force unwrap

File: CafePrices/CafePrices/Controller/MenuViewController.swift [hunk 2]
```diff
@@ -8,7 +8,7 @@
 import UIKit
 import SwiftUI
 import Firebase
-import SDWebImage
+import SDWebImage //espacios
```
Comment (hachiosaki) [inline CafePrices/CafePrices/Controller/MenuViewController.swift]: are you using sdwebimage?

File: CafePrices/CafePrices/Controller/MenuViewController.swift [hunk 3]
```diff
@@ -48,15 +49,15 @@ extension MenuViewController: UICollectionViewDelegate, UICollectionViewDataSour
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         let item = collectionView.dequeueReusableCell(withReuseIdentifier: "itemCell", for: indexPath) as? CollectionViewCell ?? CollectionViewCell()
-        let fileUrl = URL(string: items[indexPath.row].itemImage)
-        item.itemImage.sd_setImage(with: fileUrl)
-        item.itemName.text = items[indexPath.row].itemName
-        item.itemPrice.text = "$" + items[indexPath.row].itemPrice
+        let fileUrl = URL(string: items[indexPath.row].image)
+        item.itemImage.load(url: fileUrl!)
```
Comment (hachiosaki) [inline CafePrices/CafePrices/Controller/MenuViewController.swift]: dont force unwrap

File: CafePrices/CafePrices/Controller/MenuViewController.swift [hunk 4]
```diff
@@ -66,4 +67,20 @@ extension MenuViewController: UICollectionViewDelegate, UICollectionViewDataSour
      func dismiss() {
         self.dismiss(animated: true)
     }
+
```
Comment (hachiosaki) [inline CafePrices/CafePrices/Controller/MenuViewController.swift]: delete this empty line

File: CafePrices/CafePrices/SwiftUI/ItemImageSwiftUIView.swift [hunk 5]
```diff
@@ -10,18 +10,31 @@ import SDWebImageSwiftUI
...
     var body: some View {
-        let itemUrl = URL(string: object.itemImage)
-        WebImage(url: itemUrl)
+        let itemUrl = URL(string: object.image)
+        Image(systemName: "product")
+            .data(url: itemUrl!)
             .resizable()
             .aspectRatio(contentMode: .fit)
             .frame(width: 200, height: 250)
+
```
Comment (hachiosaki) [inline CafePrices/CafePrices/SwiftUI/ItemImageSwiftUIView.swift]: here are unnecessary lines, please delete

File: CafePrices/CafePrices/Controller/MenuViewController.swift [hunk 6]
```diff
@@ -8,7 +8,7 @@
 import UIKit
 import SwiftUI
 import Firebase
-import SDWebImage
+import SDWebImage //espacios
 class MenuViewController: UIViewController {
```
Comment (hachiosaki) [inline CafePrices/CafePrices/Controller/MenuViewController.swift:12]: add a space between the imports and start of the class

**Review by hachiosaki — CHANGES_REQUESTED** [review]

File: CafePrices/CafePrices/Controller/CafeteriasViewController.swift (hunk 1 above)
Comment (1699kratos) [reply CafePrices/CafePrices/Controller/CafeteriasViewController.swift]: ok

File: CafePrices/CafePrices/Controller/MenuViewController.swift (hunk 2 above)
Comment (1699kratos) [reply CafePrices/CafePrices/Controller/MenuViewController.swift]: i´ll delete it

File: CafePrices/CafePrices/Controller/MenuViewController.swift (hunk 3 above)
Comment (1699kratos) [reply CafePrices/CafePrices/Controller/MenuViewController.swift]: ok

File: CafePrices/CafePrices/Controller/MenuViewController.swift (hunk 6 above)
Comment (1699kratos) [reply CafePrices/CafePrices/Controller/MenuViewController.swift:12]: ok

File: CafePrices/CafePrices/Controller/MenuViewController.swift (hunk 4 above)
Comment (1699kratos) [reply CafePrices/CafePrices/Controller/MenuViewController.swift]: ok

File: CafePrices/CafePrices/SwiftUI/ItemImageSwiftUIView.swift (hunk 5 above)
Comment (1699kratos) [reply CafePrices/CafePrices/SwiftUI/ItemImageSwiftUIView.swift]: ok

**Review by hachiosaki — APPROVED** [review]

**Pull request merged**